product_maintenance_equipment_link/models/maintenance_equipment.py

```diff
@@ -0,0 +1,28 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields
+
+class MaintenanceEquipment(models.Model):
+    _inherit = 'maintenance.equipment'
+
+    product_ids = fields.Many2many("product.product")
+    count_products_ids = fields.Integer(compute="compute_products_count")
+
+    def compute_products_count(self):
+        for record in self:
+            record.count_products_ids = len(record.product_ids)
+    
+    def action_view_products(self):
+        action = {
```

Comment:
Could you add something to context that enables current equipment filled by default for a new product created from this link?